### PR TITLE
support undo/redo for azerty keyboards

### PR DESCRIPTION
**Pull request by Fausto95**
Fixes: https://github.com/excalidraw/excalidraw/issues/588

### CODE REVIEW

**Comment by dwelle** [comment]
We should instead migrate from [KeyboardEvent.code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) (which as this PR shows ignores locale) to [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key).

**Comment by Fausto95** [comment]
@dwelle, just migrated it to `event.key`!

**Comment by Fausto95** [comment]
oops, will migrate all of them

**Comment by dwelle** [comment]
Damn, now I recall this tweet by Marijn Haverbeke :

> Oh, wonderful. The DOM's KeyboardEvent.key property appears to be underspecified as to whether it should hold "x" or "X" for Ctrl-Shift-x and of course browsers are doing different things and good luck if you actually needed that information. [---source](https://mobile.twitter.com/MarijnJH/status/1212693616335962112)

We should investigate, first. Maybe leave the codebase rewrite for another PR, and just merge this?

**Comment by Fausto95** [comment]
So I should leave `event.code === "KeyW"` or use the `event.key` just for the undo/redo ?
@dwelle

**Comment by dwelle** [comment]
Since you've already done it, let's keep it. But I'd leave this open for a bit until someone investigates possible drawbacks. I can, but not right now.

File: src/actions/actionZindex.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const actionBringForward: Action = {
   contextItemLabel: "labels.bringForward",
   keyPriority: 40,
   keyTest: event =>
-    event[KEYS.META] && event.shiftKey && event.altKey && event.code === "KeyF",
+    event[KEYS.META] && event.shiftKey && event.altKey && event.key === "F",
```
Comment (vjeux) [inline src/actions/actionZindex.tsx]: You should remove event.shiftKey then?

**Comment by Fausto95** [comment]
> Since you've already done it, let's keep it. But I'd leave this open for a bit until someone investigates possible drawbacks. I can, but not right now.

https://caniuse.com/#feat=keyboardevent-key seems to have good browser support(latest versions).
I tested it on Firefox, Chrome & Safari, it works just fine!

File: src/actions/actionZindex.tsx (hunk 1 above)
Comment (Fausto95) [reply src/actions/actionZindex.tsx]: Yes, seems fair!

**Comment by dwelle** [comment]
I wanted to investigate the problem mentioned from the tweet above. Gonna do it in a bit, but if it doesn't reproduce on Chrome vs FF, then I'm out of luck :).

**Comment by dwelle** [comment]
Can't repro on FF vs Chrome. So maybe it is Safari. Let's merge it and when we get reports, we'll fix it then.

I'll merge with master to resolve conflicts, and merge this.

**Comment by dwelle** [comment]
Thanks @Fausto95!

**Pull request merged**